docs/Admin.md

```diff
@@ -595,6 +598,44 @@ const App = () => (
 );
 ```
 
+### Unplug the <Admin> using the `<AdminContext>` and the `<AdminUI>`
+
+Setting Resources dynamically is still very cumbersome: even if `<Admin>` accepts [a function as child](#declaring-resources-at-runtime), this function can't execute hooks.
+
+So it's impossible, for instance, to have a dynamic list of resources based on a call to the dataProvider (since the dataProvider is only defined after the Admin component renders).
+
+To do so, you have to build your own <Admin> component using both the <AdminContext> and the <AdminUI> (as we do internally). It's straightforward.
+
+``` jsx
+import * as React, { useEffect, useState } from 'react';
+import { AdminContext, AdminUI, Resource, ListGuesser, useDataProvider } from 'react-admin';
+
+function App() {
+    return (
+        <AdminContext dataProvider={myDataProvider}>
+            <AsyncResources />
+        </AdminContext>
+    );
+}
+
+function AsyncResources() {
+    const [resources, setResources] = useState([]);
+    const dataProvider = useDataProvider();
+
+    useEffect(() => {
+        dataProvider.introspect().then(r => setResources(r));
```

Comment:
introspect isn't a legal dataProvider verb, so you'll have to be more explicit here